### PR TITLE
VIT - S46878467

### PR DESCRIPTION
**Pull request by HaadiQureshi**
I had an approved late submission of 1 week, submitting on time. Regards

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Missing model details. Need to explain what the model structure is and how it works to show your understanding of it.
- Only the test accuracy plot is shown. No training accuracy and validation accuracy.

Commit messages: good, need more detail on which has been modified instead of just uploading files.

Code:
- Uses Vision transformer
- Comments mainly in the model design. Less comments in the data loader training loop and prediction file.
- Good design

Functionality/Performance:
- No patient-level split
- Model is overfitting
- Accuracy is only 0.53


General comments:
**PR is in the main branch**

- Solves the problem appropriately although performance is poor
- No patient-level split so the model is overfitting. Also need to do some hyperparameter search which hasn't been shown in the report.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem poor performance < 0.6 -1
Driver Script present
File structure present but code needs to be in own folder -2
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no patient level split -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no refs -1
Model/technical explanation missing -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in InCorrect Branch) -2
Feedback required, need to move code into own folder and use correct branch -2
Request Description minimal -1

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
